### ISSUE TITLE
test coverage for the addresses involved in a TCP server is incomplete

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5108]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5108|
|Type|defect|
|Created|2011-05-15 12:48:15Z|
|Branch|https://github.com/twisted/twisted/tree/tcp-server-address-tests-5108|

There's roughly one unit test relating to the address objects that a TCP server can find through various APIs.  `twisted.test.test_tcp.AddressTestCase` covers the three places an address might appear - passed to `buildProtocol`, returned from `getHost`, or `returned from `getPeer`.  It spreads the server coverage over two different test methods, though.

It's not a reactor builder-style test so it only covers the reactor passed to trial's command line.  It would be nice to have a separate test for each of these cases, and in a way that applies to all reactors all the time.

<details><summary>Searchable metadata</summary>

```
trac-id__5108 5108
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_tcp_server_address_tests_5108 branches/tcp-server-address-tests-5108
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1305463695000000 1305463695000000
changetime__1305473213000000 1305473213000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [31862]) Branching to 'tcp-server-address-tests-5108'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Yay tests, [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/tcp-server-address-tests-5108), please review.

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_3"></a>|@jonathanj set owner to @exarkun|
|-|-|

Just some minor things: `TCPPortTestsBuilder`, `TCPPortTestsBuilder.getListeningPort` and `TCPPortTestsBuilder.getExpectedConnectionLostLogMsg` are missing a period at the end their docstrings.

Otherwise this looks good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [31876]) Merge tcp-server-address-tests-5108

Author: exarkun
Reviewer: jonathanj
Fixes: [#5108](https://github.com/twisted/twisted/issues/5108)

Improve test coverage for the address object passed to `buildProtocol` for TCP servers
and for the return value of `ITCPTransport.getHost` and `ITCPTransport.getPeer`.
